[Flaky Test][sig-node] kubernetes-unit-test TestHTTP1DoNotReuseRequestAfterTimeout is being Flaky

### Which jobs are flaking?

periodic-kubernetes-unit-test-ppc64le

### Which tests are flaking?

1. vendor/k8s.io/client-go/rest.TestHTTP1DoNotReuseRequestAfterTimeout
2. vendor/k8s.io/client-go/rest.TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1
3. vendor/k8s.io/client-go/rest.TestHTTP1DoNotReuseRequestAfterTimeout/HTTP2

### Since when has it been flaking?

16th November 2021

### Testgrid link

https://k8s-testgrid.appspot.com/sig-node-ppc64le#unit-tests

### Reason for failure (if possible)

This test is added by change https://github.com/kubernetes/kubernetes/pull/104844

Seems like it started being flaky from the time added.

### Anything else we need to know?

Below is the trace from job:
```
 === FAIL: vendor/k8s.io/client-go/rest TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.13s)
    request_test.go:3016: Unexpected error: Get "https://127.0.0.1:34701/foo?timeout=100ms": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2021/11/20 12:06:03 http: TLS handshake error from 127.0.0.1:37890: read tcp 127.0.0.1:34701->127.0.0.1:37890: use of closed network connection
    --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.13s)
=== FAIL: vendor/k8s.io/client-go/rest TestHTTP1DoNotReuseRequestAfterTimeout (0.30s) 
```

### Relevant SIG(s)

/sig node

@aojea When I run using the stress tool the test is flaking in both ppc64le and x86_64.
- on ppc64le:
```
[root@rajalakshmi-workspace1 kubernetes]# stress ./rest.test -test.run TestHTTP1DoNotReuseRequestAfterTimeout

/tmp/go-stress-20211123T070635-1678334821
--- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout (0.45s)
    --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.28s)
        request_test.go:2974: TEST Connected from 127.0.0.1:47712 on /foo
        request_test.go:2974: TEST Connected from 127.0.0.1:47712 on /hang
        request_test.go:2976: TEST hanging 127.0.0.1:47712
        request_test.go:2974: TEST Connected from 127.0.0.1:47730 on /foo
        request_test.go:3030: Unexpected error: Get "https://127.0.0.1:35107/foo?timeout=100ms": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
FAIL


ERROR: exit status 1


/tmp/go-stress-20211123T070635-1511241200
2021/11/23 07:06:38 http: TLS handshake error from 127.0.0.1:52740: read tcp 127.0.0.1:36005->127.0.0.1:52740: read: connection reset by peer
--- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout (0.29s)
    --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.11s)
        request_test.go:3016: Unexpected error: Get "https://127.0.0.1:36005/foo?timeout=100ms": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
FAIL


ERROR: exit status 1

5s: 72 runs so far, 2 failures (2.78%)
```
- on x86_64:
```
[root@toad1 kubernetes]# stress ./rest.test -test.run TestHTTP1DoNotReuseRequestAfterTimeout
5s: 50 runs so far, 0 failures

/tmp/go-stress-20211123T025122-1432112823
--- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout (0.27s)
    --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP2 (0.14s)
        request_test.go:3016: Unexpected error: Get "https://127.0.0.1:44741/foo?timeout=100ms": context deadline exceeded
        request_test.go:2974: TEST Connected from 127.0.0.1:38406 on /foo
FAIL


ERROR: exit status 1

10s: 101 runs so far, 1 failures (0.99%)
15s: 156 runs so far, 1 failures (0.64%)
20s: 204 runs so far, 1 failures (0.49%)
25s: 250 runs so far, 1 failures (0.40%)
30s: 298 runs so far, 1 failures (0.34%)
35s: 351 runs so far, 1 failures (0.28%)
40s: 402 runs so far, 1 failures (0.25%)
45s: 454 runs so far, 1 failures (0.22%)
50s: 501 runs so far, 1 failures (0.20%)
55s: 553 runs so far, 1 failures (0.18%)
1m0s: 605 runs so far, 1 failures (0.17%)
1m5s: 655 runs so far, 1 failures (0.15%)

/tmp/go-stress-20211123T025122-2483598037
2021/11/23 02:52:32 http: TLS handshake error from 127.0.0.1:51906: read tcp 127.0.0.1:39957->127.0.0.1:51906: use of closed network connection
--- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout (0.37s)
    --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.24s)
        request_test.go:2974: TEST Connected from 127.0.0.1:51894 on /foo
        request_test.go:2974: TEST Connected from 127.0.0.1:51894 on /hang
        request_test.go:2976: TEST hanging 127.0.0.1:51894
        request_test.go:3030: Unexpected error: Get "https://127.0.0.1:39957/foo?timeout=100ms": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
FAIL


ERROR: exit status 1

1m10s: 706 runs so far, 2 failures (0.28%)
```
@aojea the golang version was
```
[root@toad1 kubernetes]# go version
go version go1.17.3 linux/amd64
[root@toad1 kubernetes]#
```

@aojea I have seen that, there are no failures after increasing the timeout to 200 Millisecond from 100 Millisecond [here](https://github.com/kubernetes/kubernetes/blob/bf9ce7fd76068903e909358d2b25b05da7e4a431/staging/src/k8s.io/client-go/rest/request_test.go#L2998) 

Before increasing the timeout:
```
[root@rajalakshmi-workspace1 kubernetes]# stress ./rest.test -test.run TestHTTP1DoNotReuseRequestAfterTimeout

/tmp/go-stress-20211129T115302-2252016254
--- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout (0.29s)
    --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.10s)
        request_test.go:2974: TEST Connected from 127.0.0.1:57294 on /foo
        request_test.go:3016: Unexpected error: Get "https://127.0.0.1:36767/foo?timeout=100ms": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
FAIL


ERROR: exit status 1

5s: 74 runs so far, 1 failures (1.35%)
10s: 158 runs so far, 1 failures (0.63%)

/tmp/go-stress-20211129T115302-927173386
--- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout (0.30s)
    --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.10s)
        request_test.go:2974: TEST Connected from 127.0.0.1:56260 on /foo
        request_test.go:3016: Unexpected error: Get "https://127.0.0.1:33675/foo?timeout=100ms": context deadline exceeded
FAIL
```

After increasing the timeout
```[root@rajalakshmi-workspace1 kubernetes]# stress ./rest.test -test.run TestHTTP1DoNotReuseRequestAfterTimeout
5s: 54 runs so far, 0 failures
10s: 112 runs so far, 0 failures
15s: 169 runs so far, 0 failures
20s: 228 runs so far, 0 failures
25s: 291 runs so far, 0 failures
30s: 353 runs so far, 0 failures
35s: 413 runs so far, 0 failures
40s: 475 runs so far, 0 failures
45s: 537 runs so far, 0 failures
50s: 598 runs so far, 0 failures
55s: 659 runs so far, 0 failures
1m0s: 721 runs so far, 0 failures
1m5s: 784 runs so far, 0 failures
1m10s: 842 runs so far, 0 failures

```

Machine information
```
[root@rajalakshmi-workspace1 kubernetes]# lscpu
Architecture:        ppc64le
Byte Order:          Little Endian
CPU(s):              8
On-line CPU(s) list: 0-7
Thread(s) per core:  8
Core(s) per socket:  1
Socket(s):           1
NUMA node(s):        1
Model:               2.2 (pvr 004e 0202)
Model name:          POWER9 (architected), altivec supported
Hypervisor vendor:   pHyp
Virtualization type: para
L1d cache:           32K
L1i cache:           32K
NUMA node0 CPU(s):   0-7
Physical sockets:    2
Physical chips:      1
Physical cores/chip: 10
```

Is there any way i can find, where its been taking longer time or why its expecting some extra time.

> @aojea I have seen that, there are no failures after increasing the timeout to 200 Millisecond from 100 Millisecond [here](https://github.com/kubernetes/kubernetes/blob/bf9ce7fd76068903e909358d2b25b05da7e4a431/staging/src/k8s.io/client-go/rest/request_test.go#L2998)
> 
> Before increasing the timeout:
> 
> ```
> [root@rajalakshmi-workspace1 kubernetes]# stress ./rest.test -test.run TestHTTP1DoNotReuseRequestAfterTimeout
> 
> /tmp/go-stress-20211129T115302-2252016254
> --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout (0.29s)
>     --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.10s)
>         request_test.go:2974: TEST Connected from 127.0.0.1:57294 on /foo
>         request_test.go:3016: Unexpected error: Get "https://127.0.0.1:36767/foo?timeout=100ms": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
> FAIL
> 
> 
> ERROR: exit status 1
> 
> 5s: 74 runs so far, 1 failures (1.35%)
> 10s: 158 runs so far, 1 failures (0.63%)
> 
> /tmp/go-stress-20211129T115302-927173386
> --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout (0.30s)
>     --- FAIL: TestHTTP1DoNotReuseRequestAfterTimeout/HTTP1 (0.10s)
>         request_test.go:2974: TEST Connected from 127.0.0.1:56260 on /foo
>         request_test.go:3016: Unexpected error: Get "https://127.0.0.1:33675/foo?timeout=100ms": context deadline exceeded
> FAIL
> ```
> 
> After increasing the timeout
> 
> ```
> 5s: 54 runs so far, 0 failures
> 10s: 112 runs so far, 0 failures
> 15s: 169 runs so far, 0 failures
> 20s: 228 runs so far, 0 failures
> 25s: 291 runs so far, 0 failures
> 30s: 353 runs so far, 0 failures
> 35s: 413 runs so far, 0 failures
> 40s: 475 runs so far, 0 failures
> 45s: 537 runs so far, 0 failures
> 50s: 598 runs so far, 0 failures
> 55s: 659 runs so far, 0 failures
> 1m0s: 721 runs so far, 0 failures
> 1m5s: 784 runs so far, 0 failures
> 1m10s: 842 runs so far, 0 failures
> ```
> 
> Machine information
> 
> ```
> [root@rajalakshmi-workspace1 kubernetes]# lscpu
> Architecture:        ppc64le
> Byte Order:          Little Endian
> CPU(s):              8
> On-line CPU(s) list: 0-7
> Thread(s) per core:  8
> Core(s) per socket:  1
> Socket(s):           1
> NUMA node(s):        1
> Model:               2.2 (pvr 004e 0202)
> Model name:          POWER9 (architected), altivec supported
> Hypervisor vendor:   pHyp
> Virtualization type: para
> L1d cache:           32K
> L1i cache:           32K
> NUMA node0 CPU(s):   0-7
> Physical sockets:    2
> Physical chips:      1
> Physical cores/chip: 10
> ```
> 
> Is there any way i can find, where its been taking longer time or why its expecting some extra time.

Maybe we can just bump the timeout in the test, is not like we need such granularity and we are taking of milliseconds only .,.. , maybe something in the standard library is racing?

--- a/staging/src/k8s.io/client-go/rest/request_test.go
+++ b/staging/src/k8s.io/client-go/rest/request_test.go
@@ -2995,7 +2995,7 @@ func TestHTTP1DoNotReuseRequestAfterTimeout(t *testing.T) {
 			config := &Config{
 				Host:      ts.URL,
 				Transport: utilnet.SetTransportDefaults(transport),
-				Timeout:   100 * time.Millisecond,
+				Timeout:   1 * time.Second,
 				// These fields are required to create a REST client.
 				ContentConfig: ContentConfig{
 					GroupVersion:         &schema.GroupVersion{},

